.github/workflows/pr-deploy-cleanup.yaml

```diff
@@ -0,0 +1,78 @@
+on:
+  pull_request:
+    types: [closed]
+jobs:
+  cleanup:
+    name: Cleanup merged deploy
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/github-script@v6
+        id: permissions
+        with:
+          script: |
+            let member;
+            try {
+              const resp = await github.rest.repos.getCollaboratorPermissionLevel({
+                username: context.actor,
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+              });
+
+              if (!resp || !resp.data || !resp.data.permission) {
+                member = false;
+              } else {
+                const permission = resp.data.permission;
+                member = permission == "admin" || permission == "write";
```

Comment:
See: https://docs.github.com/en/rest/collaborators/collaborators#get-repository-permissions-for-a-user for the list of possible permissions